src/main.cpp

```diff
@@ -627,10 +627,15 @@ int main(int argc, char** argv)
     /* Start the event listener */
     drakvuf->loop(timeout);
 
+    bool exited_on_kernel_panic = SIGDRAKVUFKERNELPANIC == drakvuf->is_interrupted();
```

Comment:
Why not just have `is_interrupted()` return the proper exit code so we can just `return` that here at the end without having to read it into a local variable first?

Reply:
Because of `drakvuf->terminate` or other logic in between may override the main interruption reason.

The second reason is that the reason for the interruption and the exit code do not have a direct mapping.

Reply:
OK, I would ask that in that case instead of custom-coding just `exited_on_kernel_panic` to do a `switch` on `is_interrupted` and return the exit codes there. I would assume there may be other situations that would warrant signaling in a similar way in the future. The default would be to just break and then return success.

Reply:
Done